Initial

Max-Pochepets opened the pull request:
HW part 1 ready for review. I tried to make the app more versitile.

Max-Pochepets committed:
Initial

tonitrus55 commented on src/main/java/core/basesyntax/App.java:
```diff
@@ -0,0 +1,40 @@
+
+public class App {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
```
Сканер не должен быть в мейне

Max-Pochepets replied:
> Сканер не должен быть в мейне

Это чисто для цикла ввода ставок от новых юзеров

tonitrus55 commented on src/main/java/core/basesyntax/App.java:
```diff
@@ -0,0 +1,40 @@
+        System.out.println(betsStorage.getStorage());
+        System.out.println(usersStorage.getStorage());
+    }
+}
```
Вообще в мейне делать логику проверки - очень плохая идея. Мейн не место для проверок.

Max-Pochepets committed:
Refactor:

little rebranding

Max-Pochepets committed:
Refactor:

ConsoleHandler.handle update

Max-Pochepets committed:
Refactor:

further improvement of ConsoleHandler.handle

dmytro-hryhoruk commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,84 @@
+    public static final int SECOND_SEQUENCE_INDEX = 1;
+
+    public void handle(Storage<User, Bet> storage) {
+        Dao<User, Bet> dao = new Dao<>(storage);
```
let's create a different dao for each entity

dmytro-hryhoruk commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,84 @@
+            System.out.println("Next? Y/N");
+            command = scanner.nextLine();
+            while (!command.equalsIgnoreCase("y") && !command.equalsIgnoreCase("n")) {
+                System.out.println("It's Y or N, you silly ^^");
```
```suggestion
                System.out.println("Enter Y or N, please");
```
It's not really friendly interface

Max-Pochepets committed:
Refactor:

message of flow continue changed to more neutral
Storage architecture changed
Dao.class changed to interface
Dao implementations for each class created
added name checks:
1. if entered name is one word only
2. if entered name is empty

dmytro-hryhoruk commented on src/main/java/core/basesyntax/App.java:
```diff
@@ -0,0 +1,15 @@
+
+        handler.handle();
+
+        System.out.println("Bets:\n" + Storage.betsStorage);
+        System.out.println("Users:\n" + Storage.usersStorage);
```
use Dao here

dmytro-hryhoruk commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,95 @@
+        }
+        if (firstName.replaceAll("[^a-zA-Z]", "").length() != firstName.length()
+                || lastName.replaceAll("[^a-zA-Z]", "").length() != lastName.length()
+                || firstName.replaceAll(" ", "").length() == 0
+                || lastName.replaceAll(" ", "").length() == 0) {
```
String has method isEmpty()

Max-Pochepets committed:
Refactor:

String.isEmpty() used
main now does not have access to DB

Max-Pochepets committed:
Merge pull request #2

* intermediate

* another checkpoint (life is hard)

* Initial

* intermediate

* another checkpoint (life is hard)

* Initial

* Merge remote-tracking branch 'origin/HW-part2' into HW-part2

* Refactor:

* Refactor:

* Refactor:

* Travis CI

* prev state Travis CI

* @Dao -> @DaoInterface
